source/base/quadrature.cc

```diff
@@ -26,11 +26,31 @@
 DEAL_II_NAMESPACE_OPEN
 
 
+namespace
+{
+  template <int dim>
+  void
+  set_tensor_basis_1d(const Quadrature<dim> &,
+                      std::vector<Quadrature<1>> &)
+  {}
```

Comment:
if you make this function return the object, rather than setting it through the second argument, then you can call it in the member initializer list of the `Quadrature` constructor. You can then also make the `tensor_basis` member variable `const`.